### PR TITLE
Bump zookeeper to latest stable version 3.7.1 in d2

### PR DESCRIPTION
**Pull request by abhilash1in**
ZK 3.4 is EOL as of 1st June, 2020 ([announcement](https://lists.apache.org/thread/xckr6nnsg9rxchkbvltkvt7hr2d0mhbo)), so this PR bumps ZK dependency to latest stable version 3.7.1. 

There should be no breaking changes between the two versions (see https://zookeeper.apache.org/releases.html).

Fixes linkedin/rest.li#810

### CODE REVIEW

File: d2/build.gradle [hunk 1]
```diff
@@ -19,6 +19,9 @@ dependencies {
   compile externalDependency.jdkTools
   compile externalDependency.zero_allocation_hashing
   compile externalDependency.xchart
+  compile externalDependency.findbugs
```
Comment (li-bsoetarman) [inline d2/build.gradle:22]: compileOnly?  or does it actually need compile

Comment (abhilash1in) [reply d2/build.gradle:22]: `compileOnly` was the first thing I tried and is technically the right option to use as we don't need findbugs during runtime, but with `compileOnly`, `./gradlew compileTestJava` still fails with the error:

```
> Task :d2-test-api:compileJava FAILED
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'value()' in type 'SuppressFBWarnings': class file for edu.umd.cs.findbugs.annotations.SuppressFBWarnings not found
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'justification()' in type 'SuppressFBWarnings'
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'value()' in type 'SuppressFBWarnings'
/Users/abkishor/.gradle/caches/modules-2/files-2.1/org.apache.zookeeper/zookeeper/3.7.1/b598708e153ff74dd2ebb800419df96df8a01c2d/zookeeper-3.7.1.jar(org/apache/zookeeper/server/ZooKeeperServer.class): warning: Cannot find annotation method 'justification()' in type 'SuppressFBWarnings'
error: warnings found and -Werror specified
1 error
4 warnings
```

The only way I could get rid of it was by using `compile`.

Comment (li-bsoetarman) [reply d2/build.gradle:22]: ah maybe you need both compileOnly and testCompileOnly.

Comment (abhilash1in) [reply d2/build.gradle]: That works, thanks! 

Updated the PR. Also had to add `compileOnly` / `testCompileOnly` to `:d2-int-test` and `:d2-test-api`.

Comment (abhilash1in) [reply d2/build.gradle]: Checks are passing as well.

**Comment by junchuanwang** [comment]
except I don't know d2 logic, this PR looks good to me

**Review by junchuanwang — APPROVED** [review]

**Review by jpstewart — APPROVED** [review]

**Pull request merged**